### PR TITLE
Refactor advanced configuration

### PR DESCRIPTION
**Pull request by adutra**
This commit introduces a new way of handling "advanced" properties:

* Instead of restricting such properties to just the store configuration, it now opens up the feature to literally any supported configuration property.
* Instead of requiring from users to input the property in its environment variable form, we now let users input the configuration property in its original form, and the conversion to the environment variable form is done internally. For example, the property `nessie.server.default-branch` is transparently transformed into the `NESSIE_SERVER_DEFAULT_BRANCH` environment variable.

### CODE REVIEW

File: helm/nessie/values.yaml [hunk 1]
```diff
@@ -70,15 +70,15 @@ postgres:
...
-versionStoreAdvancedConfig:
+# -- Advanced configuration.
+# You can pass here any valid Nessie or Quarkus configuration property.
+# Any property that is defined here takes precedence over all the other configuration values generated by this chart.
+advancedConfig:
   {}
-  # - key: NESSIE_VERSION_STORE_ADVANCED_REPOSITORY_ID
-  #   value: my-tenant
+#  nessie.server.default-branch: foo
+#  nessie.version.store.advanced.repository-id: my-tenant
```
Comment (dimas-b) [inline helm/nessie/values.yaml:79]: nit: I'd use `my-repository` as an example, to match the property meaning.

**Review by dimas-b — APPROVED** [review]

**Comment by adutra** [comment]
@dimas-b you will need to re-approve, sorry.

**Review by snazy — DISMISSED** [review]

**Review by snazy — APPROVED** [review]

**Comment by adutra** [comment]
The license/cla job seems stuck, @dimas-b or @snazy can you fix this?

**Comment by snazy** [comment]
Ah, that damn bot

**Pull request merged**